src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,468 @@
+static RadioFrequency AR62xxIdToFrequency(const uint16_t frequency_id)
+{
+  /* get frequency-part out of frequency-id */
+  unsigned frequency_encoded = frequency_id & FREQUENCY_BITMASK;
+
+  /* get channel-part out of frequency-id */
+  unsigned channel_encoded = frequency_id & CHANNEL_BITMASK;
+
+  /* calculate the frequency */
+  unsigned frequency = MIN_FREQUENCY + frequency_encoded * FREQUENCY_RANGE / MAX_FREQUENCY_ID;
+
+  /* add channel to frequency */
+  if (channel_encoded <= 3)
+    frequency += channel_encoded * 5;
+  else if (channel_encoded <= 7)
+    frequency += channel_encoded * 5 + 5;
+  else if (channel_encoded <= 11)
+    frequency += channel_encoded * 5 + 10;
+  else if (channel_encoded <= 15)
+    frequency += channel_encoded * 5 + 15;
+
+  /* generate RadioFrequency object */
+  RadioFrequency radio_frequency = RadioFrequency();
+  radio_frequency.SetKiloHertz(frequency);
+
+  return radio_frequency;
+}
+
+/**
+ * @brief converts a frequency to an AR62xx-ID
+ * 
+ * explanation: the AR62xx uses an integer value for each 8.33/25 kHz frequency starting at 118.000 MHz
+ * e.g. frequency-id 0    => 118.000 MHz
+ *      frequency-id 1    => 118.005 MHz
+ *      frequency-id 3040 => 137.000 MHz
+ * 
+ * @param frequency frequency in MHz
+ * @return uint16_t frequency-id
+ */
+static uint16_t FrequencyToAR62xxId(const RadioFrequency freq)
+{
+  /* get frequency in kiloherz */
+  unsigned frequency = freq.GetKiloHertz();
+
+  /* check if frequency is not in range */
+  if (frequency < MIN_FREQUENCY || frequency > MAX_FREQUENCY)
+    /* if the frequency is not in range we got malformed data
+        in this case we set the frequency to the first frequency possible */
+    return 0;
+
+  /* get offset of frequency to the lowest frequency which is possible */
+  unsigned frequency_offset = frequency - MIN_FREQUENCY;
+
+  /* generate frequency-id */
+  uint16_t frequency_id = frequency_offset * MAX_FREQUENCY_ID / FREQUENCY_RANGE;
+
+  /* append frequency bitmask as channel has to be added separately */
+  frequency_id &= FREQUENCY_BITMASK;
+
+  /* get channel out of frequency which is in the last two digits (which will be delivered by mod 100)  */
+  uint8_t channel = frequency % 100;
+
+  /* encode channel and add to frequency */
+  if (channel <= 15)
+    frequency_id += channel / 5;
+  else if (channel <= 40)
+    frequency_id += channel / 5 - 1;
+  else if (channel <= 65)
+    frequency_id += channel / 5 - 2;
+  else if (channel <= 90)
+    frequency_id += channel / 5 - 3;
+
+  return frequency_id;
+}
+
+/**
+ * @brief crc algorithm
+ * 
+ * @param data the data to calculate the crc on
+ * @param len the length of the data
+ * @return uint16_t the crc
+ */
+static uint16_t CRCBitwise(const uint8_t *data, const size_t len)
+{
+  uint16_t crc = 0x0000;
+  for (size_t j = len; j > 0; j--)
+  {
+    crc ^= (uint16_t)(*data++) << 8;
+    for (int i = 0; i < 8; i++)
+    {
+      if (crc & 0x8000)
+        crc = (crc << 1) ^ 0x8005;
```

Comment:
This number indicates that this is the standard CRC-16 (see https://en.wikipedia.org/wiki/Cyclic_redundancy_check).
A generic CRC algorithm doesn't belong in a driver; it is likely that another part of XCSoar will eventually need the same algorithm some day, and then we'll have several copies of the same.
Write a separate commit adding the CRC code to `util/CRC.hpp` first, and then use that library. That library already has an implementation of CRC-16-CCITT (using a precompiled lookup table).